plugins/pulp_rpm/plugins/importers/yum/pulp_solv.py

```diff
@@ -702,22 +698,21 @@ def get_units_from_solvables(self, solvables):
 
 class Solver(object):
 
-    type_factory_mapping = {
-        'rpm': rpm_unit_solvable_factory,
-        'modulemd': module_unit_solvable_factory,
-        'modulemd_defaults': module_defaults_unit_solvable_factory,
-    }
-
     def __init__(self, source_repo, target_repo=None, conservative=False):
         super(Solver, self).__init__()
         self.source_repo = source_repo
         self.target_repo = target_repo
         self.conservative = conservative
         self._loaded = False
-        self.pool = solv.Pool()
+        self._pool = solv.Pool()
         # prevent https://github.com/openSUSE/libsolv/issues/267
-        self.pool.setarch()
-        self.mapping = UnitSolvableMapping(self.pool, self.type_factory_mapping)
+        self._pool.setarch()
+        self.mapping = UnitSolvableMapping()
+        self.type_factory_mapping = {
```

Comment:
@dralley , why does it need to be an instance attribute instead of the class one?

Reply:
It doesn't, I moved it back